File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        best = 0
        seen = {}
        
        def storeLongestPath(pathL, num, numSet):
            if num in pathL or num not in numSet:
                return pathL.get(num, 0)
            val = storeLongestPath(pathL, num + 1, numSet)
            pathL[num] = val + 1
            return val + 1
            
        for num in nums:
            best = max(best, storeLongestPath(seen, num, numSet))
        return best
        

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen;
        
        for(auto num: nums){
            seen.insert(num);
        }
        
        int best = 0;
        
        for (auto num: nums){
            if(!seen.count(num - 1)){
                int cur = 0;
                while (seen.count(num)){
                    ++num;
                    ++cur;
                }
                best = max(best, cur);
            }
        }
        return best;
        
        
    }
};